cirq-core/cirq/transformers/randomized_measurements.py

```diff
@@ -13,97 +13,160 @@
+        construction of this classit defaults to measuring all the qubits in the
+        randomized bases.
 
         Args:
             circuit: The circuit to add randomized measurements to.
             rng: Random number generator.
+            unitary_ensemble: Choice of unitary ensemble (pauli/clifford/cue)
             context: Not used; to satisfy transformer API.
 
         Returns:
-            List of circuits with pre-measurement unitaries and measurements added
+            A circuit with pre-measurement unitaries and measurements added
         """
+
         if rng is None:
             rng = np.random.default_rng()
 
-        qubits = sorted(circuit.all_qubits())
-        num_qubits = len(qubits)
+        all_qubits = sorted(circuit.all_qubits())
+        if self.subsystem is None:
+            subsys_qubits = all_qubits
+        else:
+            subsys_qubits = [all_qubits[s] for s in self.subsystem]
 
-        pre_measurement_unitaries_list = self._generate_unitaries_list(rng, num_qubits)
-        pre_measurement_moment = self.unitaries_to_moment(pre_measurement_unitaries_list, qubits)
+        pre_measurement_moment = self.random_single_qubit_unitary_moment(
+            unitary_ensemble, subsys_qubits, rng
+        )
 
         return cirq.Circuit.from_moments(
-            *circuit.moments, pre_measurement_moment, cirq.M(*qubits, key='m')
+            *circuit.moments, pre_measurement_moment, cirq.M(*subsys_qubits, key="m")
         )
 
-    def _generate_unitaries_list(self, rng: np.random.Generator, num_qubits: int) -> Sequence[Any]:
-        """Generates a list of pre-measurement unitaries."""
-
-        pauli_strings = rng.choice(["X", "Y", "Z"], size=num_qubits)
-
-        if self.subsystem is not None:
-            for i in range(pauli_strings.shape[0]):
-                if i not in self.subsystem:
-                    pauli_strings[i] = np.array("Z")
-
-        return pauli_strings.tolist()
-
-    def unitaries_to_moment(
-        self, unitaries: Sequence[Literal["X", "Y", "Z"]], qubits: Sequence[Any]
-    ) -> 'cirq.Moment':
+    def random_single_qubit_unitary_moment(
+        self, unitary_ensemble: str, qubits: Sequence[Any], rng: np.random.Generator | None = None
+    ) -> "cirq.Moment":
         """Outputs the cirq moment associated with the pre-measurement rotations.
+
         Args:
-            unitaries: List of pre-measurement unitaries
+            unitary_ensemble: clifford, Pauli, cue
             qubits: List of qubits
 
-        Returns: The cirq moment associated with the pre-measurement rotations
+        Returns:
+            The cirq moment associated with the pre-measurement rotations
         """
+
+        if unitary_ensemble == "pauli":
+            unitaries = [_pauli_basis_rotation(rng) for _ in range(len(qubits))]
+
+        elif unitary_ensemble == "clifford":
+            unitaries = [_single_qubit_clifford(rng) for _ in range(len(qubits))]
+
+        elif unitary_ensemble == "cue":
```

Comment:
what does `cue` stand for?

Reply:
circular unitary ensemble. added the full form in documentation